src/components/Dialogs/Conditions/NumericCondition.js

```diff
@@ -249,48 +130,53 @@ const NumericCondition = ({
                         step={useDecimalSteps ? '0.1' : '1'}
                     />
                 )}
-            {operator && operator === OPERATOR_IN && availableLegendSets && (
-                <>
-                    <SingleSelectField
-                        selected={legendSet.id}
-                        inputWidth="136px"
-                        placeholder={i18n.t('Choose a set of options')}
-                        dense
-                        onChange={({ selected }) => {
-                            onLegendSetChange(selected)
-                            setValue(null)
-                        }}
-                    >
-                        {availableLegendSets.map((item) => (
-                            <SingleSelectOption
-                                key={item.id}
-                                value={item.id}
-                                label={item.name}
-                            />
-                        ))}
-                    </SingleSelectField>
-                    {legendSet.legends && (
-                        <MultiSelectField
-                            onChange={({ selected }) =>
-                                setValue(selected.join(';'))
-                            }
-                            inputWidth="330px"
-                            selected={(value?.length && value.split(';')) || []}
+            {operator &&
```

Comment:
The first check is not necessary when the second is `operator === OPERATOR_IN`

Reply:
Good find, removed 👍